pkg/catalog/dispatcher.go

```diff
@@ -0,0 +1,108 @@
+package catalog
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	a "github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+const (
+	maxBroadcastDeadlineTime = 15 * time.Second
+	maxGraceDeadlineTime     = 3 * time.Second
+)
+
+func (mc *MeshCatalog) dispatcher() {
+	// This will be finely tunned in near future, we can instrument other modules
+	// to take ownership of certain events, and just notify dispatcher through
+	// ScheduleBroadcastUpdate announcement type
+	subChannel := events.GetPubSubInstance().Subscribe(
+		a.ScheduleBroadcastUpdate,                                // Other modules requesting a global envoy update
+		a.ConfigMapAdded, a.ConfigMapDeleted, a.ConfigMapUpdated, // config
```

Comment:
We need to be able to contextualize this more instead of triggering envoy update on any config-map change. For ex., `enable_debug_server` doesn't need to trigger an update of envoy config.

Reply:
yeah, optimizations or taking part of these updates and handling them apart elsewhere will come after this.